pkg/distributor/distributor.go

```diff
@@ -637,6 +637,13 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 
 		if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
 			l := relabel.Process(cortexpb.FromLabelAdaptersToLabels(ts.Labels), mrc...)
+			if l == nil {
+				// all labels are gone, therefore the __name__ label is not present, metric will be discarded
+				validation.DiscardedSamples.WithLabelValues(
+					validation.DroppedByRelabelConfiguration,
+					userID,
+				).Add(float64(len(ts.Samples)))
+			}
```

Comment:
`continue` here is clearer, I think, rather than relying on a test later on.

Reply:
Done, pulled the continue to here.

Reply:
But at line 650/658 we have 

```
for _, labelName := range d.limits.DropLabels(userID) {
  removeLabel(labelName, &ts.Labels)
}
```
Assuming we don't have relabel config, but we configure Cortex to remove the  `__name__` label (and assuming the series only have `__name__` label), then we will end up with empty labels. So we should still do another `len(ts.Labels) == 0` after processing the `drop_labels` configuration right?

Reply:
I think this will never happen because if for a certain user we drop `__name__` label this means that for that user we won't ingest any metrics. But I added this if again and also created a test case to cover this scenario. I had to slightly modify the if to check for the `__name__` label as well.